src/compiler/interface.jl

```diff
@@ -61,12 +64,47 @@ end
 
 Base.adjoint(f::Function) = x -> gradient(f, x)[1]
 
+"""
+    withgradient(f, args...)
+    withgradient(f, ::Params)
+
+Returns both the value `f(args...)` and the [`gradient`](@ref), 
+`∂f/∂x` for each argument `x`, as a named tuple.
```

Comment:
```suggestion
`∂f/∂x` for each argument `x in args`/`x in Params(...)`, as a named tuple.
```

Reply:
I've tweaked this, in the end removing `∂f/∂x` here, to focus on differences from `gradient`.